src/job.js

```diff
@@ -135,37 +138,6 @@ function Job(bigQuery, id) {
      */
     get: true,
 
-    /**
```

Comment:
*The following comment is only valid if we don't need to support lazy-getting anymore*

This can be reverted, like this:

```js
getMetadata: {
  qs: {location: this.location}
}
```

Reply:
This actually seems slightly less efficient to me, if there is no location then I see no reason to create an object here. I realize the difference would be negligible, but if its all the same I see no reason to change it.

Reply:
In this PR, you had to re-create the whole `getMetadata()` function, whereas before, we inherit it from ServiceObject. It looks like you were forced to do that, because we needed to provide a `qs` object, but `ServiceObject` actually has this scenario in mind, and allows providing the `qs` object up front during instantiation.

Quick correction on my part, though:

```js
getMetadata: {
  reqOpts: {
    qs: {location: this.location}
  }
}
```

Reply:
Oh sorry, I misunderstood in my earlier comment. I think we want to continue with the lazy getting approach as mentioned previously, so this can't be reverted.